### PR TITLE
Add custom normalization option to PNG output

### PR DESCRIPTION
**Pull request by sbastrakov**
Is independent from #4164 .

- [x] History became a bit messed up, will clean it up
- [x] Merge after #4168

### CODE REVIEW

File: include/picongpu/param/png.param [hunk 1]
```diff
@@ -46,10 +46,18 @@ namespace picongpu
...
 #define EM_FIELD_SCALE_CHANNEL1 -1
 #define EM_FIELD_SCALE_CHANNEL2 -1
 #define EM_FIELD_SCALE_CHANNEL3 -1
 
+        /** SI values to be used for Custom normalization
+         *
+         * The order of normalization values is: E, B, current (note - current, not current density).
+         * This variable must always be defined, but has no effect for other normalization types.
+         */
+        constexpr float_64 customNormalizationSI[3] = {5.0e12, 5.0e12 / SI::SPEED_OF_LIGHT_SI, 15.0};
```
Comment (psychocoderHPC) [inline include/picongpu/param/png.param]: Could it be that this normalization value must be in `pic` units?

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

**Comment by psychocoderHPC** [comment]
The KHI example is shipping it's own `png.param`, this file must be updated too.

File: include/picongpu/param/png.param (hunk 1 above)
Comment (sbastrakov) [reply include/picongpu/param/png.param]: No, this is in SI, as all our input quantities are. Internally, I convert to our units inside `typicalFields<6>::get()`. I tested the implementation, these default values are for a laser with A0 = 1 (the pictures I showed you yesterday).

File: include/picongpu/plugins/output/images/Visualisation.hpp [hunk 2]
```diff
@@ -154,6 +155,21 @@ namespace picongpu
...
 
+    //! Specialization for custom normalization
+    template<>
+    struct typicalFields<6>
+    {
+        HDINLINE static float3_X get()
+        {
+            // Convert customNormalizationSI to internal units
+            using visPreview::customNormalizationSI;
+            constexpr auto normalizationE = static_cast<float_X>(customNormalizationSI[0] / UNIT_EFIELD);
```
Comment (sbastrakov) [inline include/picongpu/plugins/output/images/Visualisation.hpp]: Bug: this part should be guarded like other specializations do

File: include/picongpu/plugins/output/images/Visualisation.hpp [hunk 3]
```diff
@@ -154,6 +155,21 @@ namespace picongpu
...
+            constexpr auto normalizationB = static_cast<float_X>(customNormalizationSI[1] / UNIT_BFIELD);
+            constexpr auto normalizationCurrent
+                = static_cast<float_X>(customNormalizationSI[2] / (UNIT_CHARGE / UNIT_TIME));
+            return float3_X{normalizationE, normalizationB, normalizationCurrent};
```
Comment (sbastrakov) [inline include/picongpu/plugins/output/images/Visualisation.hpp]: Bug: E and B are in wrong order

File: include/picongpu/plugins/output/images/Visualisation.hpp (hunk 2 above)
Comment (sbastrakov) [reply include/picongpu/plugins/output/images/Visualisation.hpp]: Fixed

File: include/picongpu/plugins/output/images/Visualisation.hpp (hunk 3 above)
Comment (sbastrakov) [reply include/picongpu/plugins/output/images/Visualisation.hpp]: Fixed

**Comment by sbastrakov** [comment]
@psychocoderHPC ready for re-review. I answered the units question in conversation under it. `png.param` in all examples that use it are also updated now.

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**